pragma.md

```diff
@@ -0,0 +1,37 @@
+# pragma syntax
+## motivation
+добавляет подсказки для среды исполнения, что код может быть распараллелен или векторизован.
+
+в первом случае ускоряет выполнение за счет использования мультипоточеости без явного создания воркеров, а во втором снижает количесво обращений к памяти.
+
+> В будущем могут быть добавлены другие инструкции pragma.
+
+## high-level api
+### cycles
+```javascript
+% pragma parallel
+for (const i = 0; i < array.length; ++i) {
+  array[i] = i ** 2;
+}
+
+% pragma parallel
+array.map(e => e ** 2);
+```
+
+### short syntax
+
+```javascript
+for%parallel (const i = 0; i < array.length; ++i) {
+  array[i] = i ** 2;
+}
+
+array.map%parallel(e => e ** 2);
+```
+
+## FAQ
+### Что, если вычисление зависит не только от i-го элемента массива, но и от предыдущих (вычисленных значений)?
```

Comment:
### What if the calculation depends not only on the i-th element of the array, but also on the previous (calculated values)?
in this case, the execution is "vectorized": not one element of the array is read from memory to cache, but a certain sequence of elements, usually at least a "page" (depends on the processor and its cache).

Usually, at the beginning of each iteration of the cycle, the previous values ​​are unloaded from the cache, new ones are loaded and work is already underway with them. If this code accesses the previous values, a new memory access is sent.

Vectorization "says" processor "unload previous values ​​as rarely as possible."